### PR TITLE
DatePicker: Observe date picker input class names

### LINKED ISSUE #5121: DatePicker Validation not showing
### Discussed in https://github.com/Megabit/Blazorise/discussions/5109

<div type='discussions-op-text'>

<sup>Originally posted by **RealityMasque** October 25, 2023</sup>
Hi,

I have a validation around a DatePicker, & a custom validation method that sets the error text.  However, regardless of where the ValidationError is placed, whether or not the text is set by the method or a string in the markup, the error text does not display.  Am I missing something?

Thanks,

Orion</div>

### PR DESCRIPTION
**Pull request by stsrki**
Closes #5121

This picker makes it to observe for any changes on internal `altinput` and then to assign the validation classnames to the wrapper element.

### CODE REVIEW

**Comment by stsrki** [comment]
Test code

```
<Validations @ref="validations" Mode="ValidationMode.Manual">
    <Validation Validator="@ValidateTheDate">
        <Field Horizontal=true>
            <FieldLabel ColumnSize="ColumnSize.Is5">
                <Strong>
                    The Date
                </Strong>
            </FieldLabel>
            <FieldBody ColumnSize="ColumnSize.Is7">
                <DatePicker TValue="DateTime?" Placeholder="YYYY-MM-DD" @bind-Date="@theDate">
                    <Feedback>
                        <ValidationError />
                    </Feedback>
                </DatePicker>
            </FieldBody>
        </Field>
    </Validation>
</Validations>
<Button Color="Color.Secondary" Clicked="@(()=>ValidateTheDate())">Validate</Button>

        @code {
private Validations validations = new();

private DateTime? theDate = null;

private void ValidateTheDate( ValidatorEventArgs e )
{
    var values = e.Value as DateTime?[];
    var value = values?.Count() > 0 ? values[0] : null;

    if ( value.HasValue )
    {
        if ( value < GetEasternTimeStampDate().AddDays( 30 ) )
        {
            e.Status = ValidationStatus.Error;
            e.ErrorText = "The date must be at least 30 days in the future";
        }
        else
        {
            e.Status = ValidationStatus.Success;
        }
    }
    else
    {
        e.Status = ValidationStatus.Error;
        e.ErrorText = "Please select a date when DEV key packages are required";
    }
}

private static DateTime GetEasternTimeStampDate()
{
    return TimeZoneInfo.ConvertTimeFromUtc( DateTime.UtcNow, Eastern_Standard_Time );
}

private static TimeZoneInfo Eastern_Standard_Time = TimeZoneInfo.FindSystemTimeZoneById( "Eastern Standard Time" );

private async Task ValidateTheDate()
{
    await validations.ValidateAll();
}
        }
```

File: Source/Blazorise/wwwroot/datePicker.js [hunk 1]
```diff
@@ -317,4 +355,48 @@ export function select(element, elementId, focus) {
     if (picker && picker.altInput) {
         utilities.select(picker.altInput, null, focus);
     }
+}
+
+class ClassWatcher {
```
Comment (David-Moreira) [inline Source/Blazorise/wwwroot/datePicker.js:360]: We already have an observer. Can we make sure that the observer doesn't already do this. If it doesn't can we add this there?

File: Source/Blazorise/wwwroot/datePicker.js [hunk 2]
```diff
@@ -85,6 +85,36 @@ export function initialize(dotnetAdapter, element, elementId, options) {
                 picker.input.dispatchEvent(utilities.createEvent("input"));
             }
         });
+
+        if (options.validationStatus) {
+            const flatpickrWrapper = picker.altInput.parentElement;
+
+            if (flatpickrWrapper) {
+                if (options.validationStatus.errorClass) {
+                    function workOnClassAdd() {
```
[the remark this answers is not in the document]
Comment (stsrki) [reply Source/Blazorise/wwwroot/datePicker.js:94]: Do you mean name it `handleClassAdd()`?

File: Source/Blazorise/wwwroot/datePicker.js (hunk 1 above)
Comment (stsrki) [reply Source/Blazorise/wwwroot/datePicker.js:360]: > Can we make sure that the observer doesn't already do this.

Not sure what you mean. Can you explain?

File: Source/Blazorise/wwwroot/datePicker.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (David-Moreira) [reply Source/Blazorise/wwwroot/datePicker.js:94]: something like `classAddHandler` but maybe I'm just confusing.

File: Source/Blazorise/wwwroot/datePicker.js (hunk 1 above)
Comment (David-Moreira) [reply Source/Blazorise/wwwroot/datePicker.js:360]: image

**Pull request merged**